src/modules/70-pipeline/components/PipelineStudio/StageBuilder/StageBuilderUtil.ts

```diff
@@ -238,12 +238,26 @@ export const getFlattenedStages = (
   return { stages }
 }
 
-export const mayBeStripCIProps = (pipeline: PipelineInfoConfig): boolean => {
+export const areCIStagesAbsent = (pipeline: PipelineInfoConfig, templateTypes: { [key: string]: string }): boolean => {
   // no CI stages exist
-  const areCIStagesAbsent = pipeline?.stages?.every(stage => {
-    return stage.stage?.type !== 'CI'
+  return !!pipeline?.stages?.every(stage => {
+    const templateRef = stage.stage?.template?.templateRef
+    return stage.stage?.type !== 'CI' && (!templateRef || templateTypes[templateRef] !== 'CI')
   })
```

Comment:
nit: .some can be used for early exit to avoid iterations for pipeline with many stages
```suggestion
  return !pipeline?.stages?.some(stage => {
    const templateRef = stage.stage?.template?.templateRef
    return stage.stage?.type === 'CI' || (templateRef && templateTypes[templateRef] === 'CI')
  })
```